src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+const body = document.querySelector('body');
+const div = document.createElement('div');
+let leftClickHappened = false;
+let rightClickHappened = false;
+
+div.setAttribute('data-qa', 'notification');
+body.appendChild(div);
+
+const showNotification = (message, type) => {
+  const notification = document.createElement('div');
+
+  notification.className = type;
+  notification.setAttribute('data-qa', 'notification');
+  notification.textContent = message;
+  body.appendChild(notification);
+};
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleLeftClick = (e) => {
+    if (e.button === 0) {
+      resolve(`First promise was resolved`);
+      document.removeEventListener('mouseup', handleLeftClick);
+    }
+  };
+
+  document.addEventListener('mouseup', handleLeftClick);
+
+  setTimeout(() => {
+    reject(new Error(`First promise was rejected`));
+    document.removeEventListener('mouseup', handleLeftClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handleClick = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve(`Second promise was resolved`);
+      document.removeEventListener('mouseup', handleClick);
```

Comment:
The 'handleClick' function for 'secondPromise' removes the event listener after the promise is resolved, but there is no timeout or rejection condition. If neither button is clicked, this could potentially lead to a memory leak since the event listener would never be removed.